src/topology/sheaves/skyscraper.lean

```diff
@@ -0,0 +1,176 @@
+/-
+Copyright (c) 2022 Jujian Zhang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jujian Zhang
+-/
+import algebraic_geometry.sheafed_space
+import topology.sheaves.sheaf_condition.opens_le_cover
+import topology.sheaves.stalks
+import category_theory.preadditive.injective
+
+/-!
+# Skyscraper (pre)sheaves
+
+A skyscraper (pre)sheaf `𝓕 : (pre)sheaf C X` is the (pre)sheaf with value `A` at point `p₀` that is
+supported only at open sets contain `p₀`, i.e. `𝓕(U) = A` if `p₀ ∈ U` and `𝓕(U) = *` if `p₀ ∉ U`
+where `*` is a terminal object of `C`. In terms of stalks, `𝓕` is supported at all specializations
+of `p₀`, i.e. if `p₀ ⤳ x` then `𝓕ₓ ≅ A` and if `¬ p₀ ⤳ x` then `𝓕ₓ ≅ *`.
+
+## Main definitions
+
+* `skyscraper_presheaf`: `skyscraper_presheaf p₀ A` is the skyscraper presheaf at point `p₀` with
+  value `A`.
+* `skyscraper_sheaf`: the skyscraper presheaf satisfies the sheaf condition.
+
+## Main statements
+
+* `skyscraper_presheaf_stalk_of_specializes`: if `y ∈ closure {p₀}` then the stalk of
+  `skyscraper_presheaf p₀ A` at `y` is `A`.
+* `skyscraper_presheaf_stalk_of_not_specializes`: if `y ∉ closure {p₀}` then the stalk of
+  `skyscraper_presheaf p₀ A` at `y` is `*` the terminal object.
+
+TODO: generalize universe level when calculating stalks, after generalizing universe level of stalk.
+-/
+
+noncomputable theory
+
+section
+
+open topological_space
+open category_theory category_theory.limits
+open Top
+open opposite
+
+universes u v w
+
+variables {X : Top.{u}} (p₀ : X) {C : Type v} [category.{w} C] (A : C)
+variables [has_terminal C] [Π (U : opens X), decidable (p₀ ∈ U)]
```

Comment:
If you move the `open` and `universes` above `section`, you can remove them in the next section.
`{X : Top.{u}} (p₀ : X) [Π (U : opens X), decidable (p₀ ∈ U)]` could be moved up as well.

Reply:
done. Thanks for the suggestions